core-addon/Core.js

```diff
@@ -523,7 +542,12 @@ export default class Core {
 
       // Important: remove these lines once the migration
       // to Glean.js is finally complete.
-      await this._dataCollection.sendDeletionPing(rallyId, studyId);
+      const knownStudy = availableStudies.find(s => s.addonId == studyId);
```

Comment:
One thing I am a little concerned with here: the list of studies that the user has ever been enrolled with is stored in `activatedStudies`, so unless we also store the matching `schemaNamespace` for each as well, then this basically means we cannot remove entries from remote-settings.

We could deal with migrating this data later, `activatedStudies` is currently an array of add-on IDs so we could make this could make that an array of object `{ studyId: schemaNamespace}` objects, or something along those lines.

We could make the remote-settings format much more efficient too, it's a JSON array-of-objects right now. Having an immutable list of past and present studies might not be the worst thing.

Reply:
Hm actually I wonder if we need to do this at all, I filed https://github.com/mozilla-rally/rally-core-addon/issues/560 to discuss.

Reply:
OK per #560 let's leave the current approach in place, and not remove the pings sent on uninstall.